### PR TITLE
Add Braze client

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: basket/news/backends/braze.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        @raises: BrazeUnauthorizedError: 401 error (invalid API key)
+        @raises: BrazeForbiddenError: 403 error (invalid permissions)
+        @raises: BrazeNotFoundError: 404 error (invalid endpoint)
+        @raises: BrazeRateLimitError: 429 error (rate limit exceeded)
+        @raises: BrazeInternalServerError: 500 error (Braze server error)
+        @raises: BrazeClientError: any other error
+
+        """
+        if not self.active:
+            return
```
Comment (stevejalim) [inline basket/news/backends/braze.py:76]: Question, non-blocking: do we want to log anything to flag this in logs?

File: basket/news/backends/braze.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+        """
+        data = {
+            "user_aliases": [{"alias_name": email, "alias_label": "email"}],
+            "email_address": email,
+        }
+
+        return self._request(BrazeEndpoint.USERS_EXPORT_IDS, data)
+
+    def send_campaign(self, email, campaign_id):
```
Comment (stevejalim) [inline basket/news/backends/braze.py:158]: Question, non-blocking: do we need a bulk version of this as well?

**Review by stevejalim — COMMENTED** [review]
r+ wq

📤

**Review by stevejalim — APPROVED** [review]

File: basket/news/backends/braze.py (hunk 1 above)
Comment (robhudson) [reply basket/news/backends/braze.py:76]: There's a one-time warning on start up if the API key isn't supplied that should suffice. I just wanted it to be callable but no-op so the code integration can exist without throwing an error.

File: basket/news/backends/braze.py (hunk 2 above)
Comment (robhudson) [reply basket/news/backends/braze.py:158]: These are just transactional emails even though we are using the campaign API, so shouldn't be called in bulk. Kinda confusing, but apparently transactional emails are an add-on.

**Pull request merged**